velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -123,163 +146,169 @@ class ParquetTpchTest : public testing::Test {
+        readerOptions);
+    auto type = reader.typeWithId();
+    auto names = type->type->as<TypeKind::ROW>().names();
+    std::vector<std::shared_ptr<const Type>> types;
+    for (int i = 0; i < type->size(); i++) {
+      types.push_back(createScalarType(type->childAt(i)->type->kind()));
     }
     return ROW(std::move(names), std::move(types));
   }
 
+  std::shared_ptr<const facebook::velox::RowType> getRowType(
+      const std::string& table,
+      const std::vector<std::string>& columns,
+      const std::string& path) {
+    auto columnSelector =
+        std::make_shared<facebook::velox::dwio::common::ColumnSelector>(
+            getRowType(table, path), columns);
+
+    return std::move(columnSelector->buildSelectedReordered());
+  }
+
   std::shared_ptr<Task> assertQuery(
       const CursorParameters& params,
-      const std::string& filePath,
-      const core::PlanNodeId& sourcePlanNodeId,
+      const std::vector<std::string>& filePaths,
+      const std::vector<core::PlanNodeId>& sourcePlanNodeIds,
       const std::string& duckQuery) const {
     bool noMoreSplits = false;
     return exec::test::assertQuery(
         params,
         [&](exec::Task* task) {
           if (!noMoreSplits) {
-            auto const& splits = makeSplits(filePath);
-            for (const auto& split : splits) {
-              task->addSplit(sourcePlanNodeId, exec::Split(split));
+            for (size_t i = 0; i < filePaths.size(); i++) {
+              auto const& splits = makeSplits(filePaths[i]);
+              for (const auto& split : splits) {
+                task->addSplit(sourcePlanNodeIds[i], exec::Split(split));
+              }
+              task->noMoreSplits(sourcePlanNodeIds[i]);
+              noMoreSplits = true;
             }
-            task->noMoreSplits(sourcePlanNodeId);
-            noMoreSplits = true;
           }
         },
         duckQuery,
         *duckDb_);
   }
 
   static std::shared_ptr<DuckDbQueryRunner> duckDb_;
-  std::shared_ptr<exec::test::TempDirectoryPath> tempDirectory_;
-  RowTypePtr lineitemType_;
+  static std::shared_ptr<exec::test::TempDirectoryPath> tempDirectory_;
+  static std::map<std::string, std::string> parquetFilesPaths_;
+  static std::shared_ptr<PlanNodeIdGenerator> planNodeIdGenerator_;
 };
 
 std::shared_ptr<DuckDbQueryRunner> ParquetTpchTest::duckDb_ = nullptr;
+std::shared_ptr<exec::test::TempDirectoryPath> ParquetTpchTest::tempDirectory_ =
+    nullptr;
+std::shared_ptr<PlanNodeIdGenerator> ParquetTpchTest::planNodeIdGenerator_ =
+    nullptr;
+std::map<std::string, std::string> ParquetTpchTest::parquetFilesPaths_;
 
 TEST_F(ParquetTpchTest, q1) {
-  const auto filePath = writeLineitemTableToParquet();
-
-  auto rowType = getLineitemColumns(
-      {"returnflag",
-       "linestatus",
-       "quantity",
-       "extendedprice",
-       "discount",
-       "tax",
-       "shipdate"});
+  const auto& filePath = parquetFilesPaths_["lineitem"];
+  TableColumnsMap tableCols;
```

Comment:
This map contains only only entry with key "lineitem". You could use and std::vector<std::string> intead.

```
std::vector<std::string> columnNames = {...};
```

do not abbreviate: tableCols -> columnNames